### DISCUSSION TITLE
Facing issue to load after finetuning

### DISCUSSION THREAD

**Discussion by EnugulaVishnucharan** [SalesforceAIResearch/uni2ts#93] [Q&A]
I am trying to finetune moirai model on my data . After finetuning with the script given in the readme i,e 
python -m cli.train \
 -cp conf/finetune \
 run_name=example_run \ 
 model=moirai_1.0_R_small \ 
 data=etth1 \ 
 val_data=etth1

It is producing the files attached . 
![Screenshot from 2024-07-27 11-58-50](https://github.com/user-attachments/assets/b720c262-82e4-4d74-82f9-af54448f1648)
But when I load the finetuned model will full path 
model = MoiraiForecast(
    module=MoiraiModule.from_pretrained("fullpathofthefinetuneddirectory"),
    prediction_length=PDT,
    context_length=CTX,
    patch_size=PSZ,
    num_samples=100,
    target_dim=1,
    feat_dynamic_real_dim=ds.num_feat_dynamic_real,
    past_feat_dynamic_real_dim=ds.num_past_feat_dynamic_real,
    module_kwargs=checkpoint['hyper_parameters']['module_kwargs']
)
it is giving 
MoiraiModule.init() missing 7 required positional arguments: 'distr_output', 'd_model', 'num_layers', 'patch_sizes', 'max_seq_len', 'attn_dropout_p', and 'dropout_p'

I could find there arguments in the checkpoints and also the config.yaml files in the output finetuned model folder.

Please let me know where am I missing

**Comment by gorold** [comment]
You can use a config file like this https://github.com/SalesforceAIResearch/uni2ts/blob/7f9a34ec5b6a55d831baa54aca77e80044706277/cli/conf/eval/model/moirai_lightning_ckpt.yaml

**Reply by EnugulaVishnucharan** [reply]
Thank you for the response. Able to forecast now.
